dags/utils/gcp.py

```diff
@@ -505,6 +505,20 @@ def bigquery_etl_query(
         **kwargs
     )
 
+    publish_json_operator = publish_json(
```

Comment:
Am I correct in understanding that this will add a publish_json operator for every query but in most cases it simply checks the metadata file, determines it's not public, and shuts down?

If so, that's going to add a lot of noise to the DAGs and potentially have performance implications for tracking all the extra tasks. For the time being, I think it would be preferable to add publish_json explicitly for each public table, and we defer having a fully-metadata-based solution until we're generating DAGs in bigquery-etl.

Reply:
A simpler alternative is add an argument here for publishing json data.

Reply:
Right, it would probably be more convenient and intuitive if users would not need to set the `public_json` flag in the metadata file and additionally change the task in Airflow to set the flag. So, I think the better way for now is to execute both scripts in the bigquery-etl entrypoint (at least until the DAG generation stuff is in place)

Reply:
On second thought, if publishing json is part of the `bigquery_etl_query` tasks, then we'd need to use the highmem node pool for all of them. Those machines seem to be a bit more expensive. So, maybe in the meantime setting an argument for publishing json is the way to go. I suspect there won't be too many use cases for datasets to be published until the lower-friction scheduled queries get implemented (which should happen before/by the end of H1)